yjit/src/codegen.rs

```diff
@@ -6068,7 +6086,9 @@ fn gen_send_cfunc(
     if kw_arg.is_null() && flags & VM_CALL_OPT_SEND == 0 && flags & VM_CALL_ARGS_SPLAT == 0 && (cfunc_argc == -1 || argc == cfunc_argc) {
         let expected_stack_after = asm.ctx.get_stack_size() as i32 - argc;
         if let Some(known_cfunc_codegen) = lookup_cfunc_codegen(unsafe { (*cme).def }) {
-            if perf_call!("gen_send_cfunc: ", known_cfunc_codegen(jit, asm, ocb, ci, cme, block, argc, recv_known_class)) {
+            if asm.with_leaf_ccall(|asm|
+                perf_call!("gen_send_cfunc: ", known_cfunc_codegen(jit, asm, ocb, ci, cme, block, argc, recv_known_class))
+            ) {
```

Comment:
Here, you are assuming that by default, all known cfunc codegen will be leaf? I guess this tracks current usage?

Reply:
Correct. The interpreter skips pushing a frame for `opt_*` insns (e.g. `String#<<` implemented by `jit_rb_str_concat`), and I want to allow YJIT to do the same thing in that case, but they should be leaf by default.

Reply:
Could maybe add a comment to explicitly say "We assume that specialized cfunc codegen will only be done for methods that are leaf".

Reply:
added a comment here https://github.com/ruby/ruby/pull/10002/commits/60f635961dbd425d45b01a5647c515fd4edca84e